datalab/datalab_session/data_operations/normalization.py

```diff
@@ -38,26 +38,22 @@ def wizard_description():
                 }
             }
         }
-    
-    def operate(self):
 
+    def operate(self):
         input_list = self.input_data.get('input_files', [])
         log.info(f'Normalization operation on {len(input_list)} file(s)')
 
-        input_fits_list = []
-        for index, input in enumerate(input_list, start=1):
-            input_fits_list.append(InputDataHandler(input['basename'], input['source']))
-            self.set_operation_progress(0.5 * (index / len(input_list)))
-
         output_files = []
-        for index, image in enumerate(input_fits_list, start=1):
-            median = np.median(image.sci_data)
-            normalized_image = image.sci_data / median
-
-            comment = f'Datalab Normalization on file {input_list[index-1]["basename"]}'
-            output = FITSOutputHandler(f'{self.cache_key}', normalized_image, comment).create_and_save_data_products(index=index)
-            output_files.append(output)
-            self.set_operation_progress(0.5 + index/len(input_fits_list) * 0.4)
+        for index, input in enumerate(input_list, start=1):
```

Comment:
Changed the ordering of this operation so it acts on the input images sequentially and cleans up memory between image. This should allow it to operate on large numbers of images without running out of memory.